test/debug_info/entry_argument_1.f90

```diff
@@ -0,0 +1,18 @@
+!RUN: %flang -g -S -emit-llvm %s -o - | FileCheck %s
+
+!CHECK: [[N1:![0-9]+]] = distinct !DISubprogram
+!CHECK: !DILocalVariable(arg: 1, scope: [[N1]]
+!CHECK: !DILocalVariable(arg: 2, scope: [[N1]]
+!CHECK: !DILocalVariable(name: "a", arg: 3, scope: [[N1]]
+
+module test
+contains
+  subroutine sub(a)
+    implicit none
+    integer(kind = 4) :: m
+    real(kind = 8),intent(inout) :: a(:,:)
+    m = size(a, 1)
+    entry subsub(a)
```

Comment:
What happens if there are more entries in the subroutine? Can you add another `entry` declaration and the corresponding `CHECK` pattern to this test case?

Reply:
For the entry feature, flang's processing logic is to turn an ordinary function into a function with parameter selection, and jump to which label (entry) according to the selection.
Therefore, adding one more entry will not affect the argument's metadata.

Reply:
``For entry declaration,the complier will turn an ordinary function into a function with branch, and jumping to which label (entry) is according to the choice argument.
Therefore, adding one more entry will not affect the argument's metadata, and that only to add branch label.
Like this:
`define void @test__master___test_sub__(i32 %__master_entry_choice430.arg, ptr %a, ptr %a$sd)`
......
`br i1 %3, label %Ltest_subsub_, label %L.LB2_432`
......
`br i1 %5, label %Ltest_subsub1_, label %L.LB2_433`

Reply:
I added another entry declaration in test case

Reply:
OK thanks.